ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d.hpp

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/core.hpp"
+#include "ttnn/types.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/run_operation.hpp"
+#include "ttnn/tensor/host_buffer/functions.hpp"
+
+#include "device/max_pool2d_device_op.hpp"
+
+
+namespace ttnn {
+namespace operations::pool {
+
+struct MaxPoolNewOp {
+    template<typename T>
+    static Tensor operator()(uint8_t queue_id, const Tensor& input_tensor, uint32_t batch_size, uint32_t input_h, uint32_t input_w, uint32_t channels, std::array<uint32_t, 2> kernel_size, std::array<uint32_t, 2> stride, std::array<uint32_t, 2> padding, std::array<uint32_t, 2> dilation, T* device);
```

Comment:
This is not typical for a ttnn operation today, but we discussed with the team recently that we expect ops to potentially gain this parameter overtime.

@mywoodstock  can you share more about this argument `T* device`? 

CC @eyonland @arakhmati @dmakoviichuk-tt

Reply:
Yes, so maxpool (and conv as well), use the halo construction infra which need the device to obtain the physical noc coords to be used in the op -- and it is template'ized to facilitate multi-device use.